local-test-configuration/scripts/initialize-chain.ts

```diff
@@ -112,64 +143,92 @@ const loadAirnodeFeedConfig = (airnodeFeedDir: 'airnode-feed-1' | 'airnode-feed-
+    const updateParameters = encodeUpdateParameters();
+
+    let sponsorAddressHash;
+    switch (rawConfig.walletDerivationScheme.type) {
+      case 'self-funded': {
+        sponsorAddressHash = deriveSponsorAddressHashForSelfFundedFeed(dapiName, updateParameters);
+        break;
+      }
+      case 'managed': {
+        sponsorAddressHash = deriveSponsorAddressHashForManagedFeed(dapiName);
+        break;
+      }
+      case 'fixed': {
+        sponsorAddressHash = rawConfig.walletDerivationScheme.sponsorAddress;
+        break;
+      }
+    }
     const sponsorWallet = deriveSponsorWalletFromSponsorAddressHash(airseekerWalletMnemonic, sponsorAddressHash);
-    const sponsorWalletBalance = await funderWallet.provider!.getBalance(sponsorWallet.address);
+    const sponsorWalletAddress = await sponsorWallet.getAddress();
+    const sponsorWalletBalance = await funderWallet.provider!.getBalance(sponsorWalletAddress);
     console.info('Sponsor wallet balance:', ethers.formatEther(sponsorWalletBalance.toString()));
 
     const tx = await funderWallet.sendTransaction({
-      to: sponsorWallet.address,
-      value: ethers.parseEther('1'),
+      to: sponsorWalletAddress,
+      value: ethers.parseEther('0.1'),
     });
     await tx.wait();
 
     console.info(`Funding sponsor wallets`, {
       dapiName,
-      sponsorWalletAddress: sponsorWallet.address,
+      decodedDapiName: ethers.decodeBytes32String(dapiName),
+      sponsorWalletAddress,
     });
   }
 };
 
-export const deploy = async (funderWallet: ethers.HDNodeWallet, provider: ethers.JsonRpcProvider) => {
+export const deploy = async (funderWallet: ethers.NonceManager, provider: ethers.JsonRpcProvider) => {
   // NOTE: It is OK if all of these roles are done via the funder wallet.
-  const deployerAndManager = funderWallet,
-    randomPerson = funderWallet;
+  const deployerAndManager = funderWallet;
+  const deployerAndManagerAddress = await deployerAndManager.getAddress();
+
+  const randomPerson = ethers.Wallet.createRandom().connect(deployerAndManager.provider);
+  const fundRandomPersonTx = await deployerAndManager.sendTransaction({
+    to: await randomPerson.getAddress(),
+    value: ethers.parseEther('1'),
+  });
+  await fundRandomPersonTx.wait();
 
   // Deploy contracts
-  const accessControlRegistryFactory = new AccessControlRegistryFactory(deployerAndManager as Signer);
+  const accessControlRegistryFactory = new AccessControlRegistryFactory(deployerAndManager);
   const accessControlRegistry = await accessControlRegistryFactory.deploy();
   await accessControlRegistry.waitForDeployment();
-  const api3ServerV1Factory = new Api3ServerV1Factory(deployerAndManager as Signer);
+  const api3ServerV1Factory = new Api3ServerV1Factory(deployerAndManager);
   const api3ServerV1AdminRoleDescription = 'Api3ServerV1 admin';
   const api3ServerV1 = await api3ServerV1Factory.deploy(
-    accessControlRegistry.getAddress(),
+    await accessControlRegistry.getAddress(),
```

Comment:
Not really. I just wanted to be consistent since BaseContract and BaseWallet both inherit from Addresseable. I'll revert to just use `.address` property on Wallets

Reply:
I was OK with `await` ing here, but was curious if there is some particular reason for it.

Reply:
Sorry, that answer was actually about using the `async getAddress()` over the `address` property. The deploy() function accepts either `string` or `Promise<string>` like you mentioned so I just reverted the change.